src/basic/capability-util.c

```diff
@@ -370,3 +370,18 @@ int drop_capability(cap_value_t cv) {
 
         return 0;
 }
+
+bool ambient_capabilities_supported(void) {
+        static int cache = -1;
+
+        if (cache >= 0)
+                return cache;
+
+        /* If PR_CAP_AMBIENT returns something valid, or an unexpected error code we assume that ambient caps are
+         * available. */
+
+        cache = prctl(PR_CAP_AMBIENT, PR_CAP_AMBIENT_IS_SET, CAP_KILL, 0, 0) >= 0 ||
+                !IN_SET(errno, EINVAL, EOPNOTSUPP, ENOSYS);
```

Comment:
Because system calls can fail for many different reasons (for example, some seccomp or selinux thing), and I wanted to hence make sure we only consider ambient caps to be unavailable if the error code returned is actually one of the "not supported" ones, which the three above are. (prctl usually reports unavailable options with EINVAL, and EOPNOTSUPP and ENOSYS are usually how this is reported).

This way, if ambient caps fail with EPERM we will still consider them to be available, which then means we'll try to use them later on, where we then will see EPERM again and report it as usual...

Or to say this differently: this function is not supposed to process errors or anything, it shall simple report back if the ambient caps API is implemented or not.